Pod/Classes/KolodaView/SwipeResultDirection.swift

```diff
@@ -0,0 +1,183 @@
+    case Left
+    case Right
+    case Up
+    case Down
+    case TopLeft
+    case TopRight
+    case BottomLeft
+    case BottomRight
+}
+
+extension SwipeResultDirection {
+    
+    private var swipeDirection:Direction {
+        switch self{
+        case .Up: return .Up
+        case .Down: return .Down
+        case .Left: return .Left
+        case .Right: return  .Right
+        case .TopLeft: return .TopLeft
+        case .TopRight: return .TopRight
+        case .BottomLeft: return .BottomLeft
+        case .BottomRight: return .BottomRight
+        case .None: return .None
+        }
+    }
+    
+    var point:CGPoint {
+        return self.swipeDirection.point
+    }
+    
+    var bearing:Double {
+        return self.swipeDirection.bearing
+    }
+    
+    static var boundsRect:CGRect {
+        let w = HorizontalPosition.Right.rawValue - HorizontalPosition.Left.rawValue
+        let h = VerticalPosition.Bottom.rawValue - VerticalPosition.Top.rawValue
+        return CGRect(x: HorizontalPosition.Left.rawValue, y: VerticalPosition.Top.rawValue, width: w, height: h)
+    }
+}
+
+
+private enum VerticalPosition:CGFloat {
+    case Top = -1
+    case Middle = 0
+    case Bottom = 1
+}
+
+private enum HorizontalPosition:CGFloat {
+    case Left = -1
+    case Middle = 0
+    case Right = 1
+}
+
+
+private struct Direction {
+    let horizontalPosition:HorizontalPosition
+    let verticalPosition:VerticalPosition
+    
+    var point:CGPoint {
+        return CGPoint(x:horizontalPosition.rawValue, y: verticalPosition.rawValue)
+    }
+    
+    var bearing:Double {
+        return self.point.bearingTo(Direction.None.point)
+    }
+    
+    static let None = Direction(horizontalPosition: .Middle, verticalPosition: .Middle)
+    static let Up = Direction(horizontalPosition: .Middle, verticalPosition: .Top)
+    static let Down = Direction(horizontalPosition: .Middle, verticalPosition: .Bottom)
+    static let Left = Direction(horizontalPosition: .Left, verticalPosition: .Middle)
+    static let Right = Direction(horizontalPosition: .Right, verticalPosition: .Middle)
+    
+    static let TopLeft = Direction(horizontalPosition: .Left, verticalPosition: .Top)
+    static let TopRight = Direction(horizontalPosition: .Right, verticalPosition: .Top)
+    static let BottomLeft = Direction(horizontalPosition: .Left, verticalPosition: .Bottom)
+    static let BottomRight = Direction(horizontalPosition: .Right, verticalPosition: .Bottom)
+}
+
+
+//MARK: Geometry
+
+extension CGPoint {
+    func distanceTo(point:CGPoint) -> CGFloat {
+        return sqrt(pow(point.x - self.x, 2) + pow(point.y - self.y, 2))
+    }
+    
+    func bearingTo(point:CGPoint) -> Double {
+        return atan2(Double(point.y - self.y), Double(point.x - self.x))
+    }
+    
+    func scalarProjectionWith(point:CGPoint) -> CGFloat {
+        return dotProductWith(point)/point.modulo
+    }
+    
+    func scalarProjectionPointWith(point:CGPoint) -> CGPoint {
```

Comment:
Ok there was actually a bug in that created point's `y` needed to be multiplied by `r` and not `y`.
But the purpose of this method is to calculate the point C in the figure below:
![untitled](https://cloud.githubusercontent.com/assets/1863251/14570245/2041951c-02f8-11e6-85f4-e5668f0e4dcd.png)